docs/tutorials/basics/70_blocks.md

```diff
@@ -0,0 +1,570 @@
+# Blocks
+
+As we already mentioned in the [methods](./60_methods.md#blocks) section, *blocks* is an interesting and important topic. So let's start defining them:
+
+A *block of code* or *block* for short, is a way to abstract a sequence of interactions between objects. They are commonly used as a method parameter, as a way to parameterize (forgive the redundancy) part of its behavior.
+
+## Blocks and Methods
+
+Methods may receive a *block* as an argument. And using the keyword `yield` indicates the place in the method's body where we want to "place" said `block`:
+
+```crystal-play
+def with_42
+  yield
+end
+
+with_42 do
+  puts 42
+end
+```
+
+We can also use `yield` in more than one place:
+
+```crystal-play
+def three_times
+  yield
+  yield
+  yield
+end
+
+three_times do
+  puts 42
+end
+```
+
+Let's see another examples with a common use scenario for *blocks*: collections.
+
+Here are two examples:
+
+```crystal-play
+# Example 1
+arr = [1, 2, 3]
+index = 0
+
+while index < arr.size
+  elem = arr[index]
+
+  puts elem + 42
+
+  index = index + 1
+end
+```
+
+```crystal-play
+# Example 2
+arr = ["John", "Paul", "George", "Ringo"]
+index = 0
+
+while index < arr.size
+  elem = arr[index]
+
+  puts "Hello #{elem}"
+
+  index = index + 1
+end
+```
+
+As we can see in the above examples, the structures are the same: we want to traverse the array and do *something* with the current element at each step. And here is where *blocks* come handy: we can parameterize that *something* we want to execute.
+
+Now, let's rewrite the examples. We are going to declare a new method that will abstract the structure duplicated in both examples: It will receive the array and what we want to do at each step using a *block*:
```

Comment:
```suggestion
Now, let's rewrite the examples. We are going to declare a new method that will abstract the structure duplicated in both examples: It will receive the array and what we want to do at each step using a block:
```